request/consumer.go

```diff
@@ -0,0 +1,49 @@
+package request
+
+import (
+	"github.com/ONSdigital/dp-observation-extractor/kafka"
+	"github.com/ONSdigital/dp-observation-extractor/schema"
+	"github.com/ONSdigital/go-ns/log"
+)
+
+// MessageConsumer provides a generic interface for consuming []byte messages
+type MessageConsumer interface {
+	Incoming() chan kafka.Message
+	Closer() chan bool
+}
+
+// Handler represents a handler for processing a single request.
+type Handler interface {
+	Handle(request *Request) error
+}
+
+// Consume convert them to request instances, and pass the request to the provided handler.
+func Consume(messageConsumer MessageConsumer, handler Handler) {
+	for message := range messageConsumer.Incoming() {
+
+		request, err := Unmarshal(message)
```

Comment:
Should the request be referred to as an event?

Something has happened and we want to do something with this information (event), not that some service has told us to do something (request).